plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2306,8 +2312,10 @@ void producer_plugin_impl::retire_expired_deferred_trxs(const fc::time_point& de
    auto               expired_itr         = expired_idx.begin();
 
    while (expired_itr != expired_idx.end()) {
-      if (expired_itr->expiration >= pending_block_time) {
-         break; // not expired yet
+      // * Before disable_deferred_trxs_stage_1 is activated, retire only expired deferred trxs.
+      // * After disable_deferred_trxs_stage_1, retire any deferred trxs in any order
+      if (!chain.is_builtin_activated( builtin_protocol_feature_t::disable_deferred_trxs_stage_1) && expired_itr->expiration >= pending_block_time) { // before stage_1 and not expired yet
```

Comment:
Should cache the value of `chain.is_builtin_activated( builtin_protocol_feature_t::disable_deferred_trxs_stage_1)` outside the while loop.